R/diet.R

```diff
@@ -0,0 +1,133 @@
+#'   \item -2 points for no carrot intake
+#'   \item -2 points per daily frquency of fruit juce consumption greater than 
+#'   once/day (maximum -10 points)
+#'  }
+#'
+#' @param FVCDFRU daily consumption of fruit
+#'
+#' @param FVCDSAL daily consumption of green salad
+#'
+#' @param FVCDPOT daily consumption of potatoes
+#'
+#' @param FVCDCAR daily consumption of carrots
+#'
+#' @param FVCDVEG daily consumption of other vegetables
+#'
+#' @param FVCDJUI daily consumption of fruit juice
+#' 
+#' @param DHH_SEX sex; 1 = male, 2 = female
+#'
+#' @example
+#' # Using the 'diet_score_fun' function to create the derived diet variable  
+#' # across CCHS cycles.
+#' # diet_score_fun() is specified in the variable_details.csv
+#'
+#' # To create a harmonized diet_score variable across CCHS cycles, use 
+#' # rec_with_table() for each CCHS cycle and specify diet_score_fun and the
+#' # required base variables.
+#' # Using bind_rows(), you can combine smoke_simple across cycles
+#'
+#' library(cchsflow)
+#'
+#' diet_score2009_2010 <- rec_with_table(
+#'   cchs2009_2010_p, c(
+#'     "FVCDFRU", "FVCDSAL", "FVCDPOT", "FVCDCAR", "FVCDVEG", "FVCDJUI", 
+#'     "DHH_SEX", "diet_score_fun"
+#'   )
+#' )
+#'
+#' head(diet_score2009_2010)
+#'
+#' diet_score2011_2012 <- rec_with_table(
+#'   cchs2011_2012_p,c(
+#'     "FVCDFRU", "FVCDSAL", "FVCDPOT", "FVCDCAR", "FVCDVEG", "FVCDJUI", 
+#'     "DHH_SEX", "diet_score_fun"
+#'   )
+#' )
+#'
+#' tail(diet_score2011_2012)
+#'
+#' combined_diet_score <- suppressWarnings(bind_rows(diet_score2009_2010,
+#'  diet_score2011_2012))
+#'
+#' head(combined_diet_score)
+#' tail(combined_diet_score)
+#' @export
+diet_score_fun <-
+  function(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, FVCDVEG, FVCDJUI, DHH_SEX) {
+    
+    # Nested function: total fruit and vegetables, excluding fruit juice
+    derive_total_fruitveg <- function(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
+                                      FVCDVEG) {
+      total_fruitveg <- 
+        if_else2(!is.na(FVCDFRU) & !is.na(FVCDSAL) & !is.na(FVCDPOT) & 
+                   !is.na(FVCDCAR)  & !is.na(FVCDVEG), FVCDFRU + FVCDSAL + 
+                   FVCDPOT + FVCDCAR + FVCDVEG, NA)
+      return(total_fruitveg)
+    }
+    total_fruitveg <- derive_total_fruitveg(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
+                                            FVCDVEG)
+    
+    # Nested function: maximum total fruit and vegetables = 8
+    derive_max_fruitveg <- function(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
+                                    FVCDVEG) {
+      max_fruitveg <-
+        if_else2(total_fruitveg >8, 8,
+               if_else2(!is.na(total_fruitveg), total_fruitveg, NA))
+      return(max_fruitveg)
+    }
+    max_fruitveg <- derive_max_fruitveg(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
+                                            FVCDVEG)
+    
+    # Nested function: high potato intake flag
+    potato_fun <- function(FVCDPOT, DHH_SEX) {
+      FVCDPOT_high <-
```

Comment:
This code should be split up into 2 parts to avoid checking for DHH_SEX twice. The first part get daily limit to use depending on the sex and the second part sets the high flag based on that calculated daily limit.